File: 1.Array/5.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums){
    let count = 0
    let candidate = null

    for(let num of nums){
        if(count === 0){
            candidate = num
        }

        if(num === candidate){
            count++
        }else{
            count--
        }
    }

    return candidate
}

// Driven code
// Input: nums = [2,2,1,1,1,2,2]
// Output: 2

let nums = [2,2,1,1,1,2,2]

console.log(majorityElement(nums))

File: 1.Array/5.py
def majorityElement(nums):
    count = 0
    candidate = None

    for num in nums:
        if count == 0:
            candidate = num
        if num == candidate:
            count += 1
        else:
            count -= 1
    return candidate

nums = [2,2,1,1,1,2,2]

print(majorityElement(nums))